aten/src/ATen/core/function_schema_inl.h

```diff
@@ -121,17 +121,40 @@ inline bool FunctionSchema::isBackwardCompatibleWith(
     }
   }
 
-  // Make sure that all the old arguments have their corresponding backward
-  // compatible arguments in this schema.
-  for (size_t i = 0; i < old.arguments().size(); ++i) {
+  // we want to test both out and default args seperately
+
+  // find the start of out args in the old schema
+  int old_out_start_idx;
```

Comment:
Future clean up:
It feels like that this loop has appeared often enough to warrant a utility method, what do you think?

Reply:
will di